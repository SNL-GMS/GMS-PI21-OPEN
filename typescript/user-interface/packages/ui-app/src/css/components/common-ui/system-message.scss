$critical-selector: '[data-severity*=CRITICAL]';
$warning-selector: '[data-severity*=WARNING]';
$info-selector: '[data-severity*=INFO]';

@mixin createColoredCount($selector, $color) {
  #{$selector} &__count {
    color: lighten($color, 10%);
  }
}

.system-message-display {
  @include flexColumn;
  width: 100%;
  height: 100%;
  .toolbar {
    margin-bottom: 0.5em;
  }
}

.system-message-table {
  @include flexColumn;
  width: 100%;
  height: 100%;
  background-color: var(--gms-prominent-background, $gms-prominent-background);
  .system-cell {
    &#{$critical-selector} {
      background-color: var(--system-message-critical-color, $system-message-critical-color);
    }
    &#{$warning-selector} {
      background-color: var(--system-message-warning-color, $system-message-warning-color);
    }
    &#{$info-selector} {
      background-color: var(--system-message-info-color, $system-message-info-color);
    }
  }
  &__button {
    &--floating {
      position: absolute;
      left: 50%;
      bottom: 3rem;
      transform: translate3D(-50%, -50%, 0);
    }
    &--desc {
      bottom: auto;
      top: 9rem;
    }
  }
}

.system-message-summary {
  @include centerChildren;
  background: var(--gms-prominent-background, $gms-prominent-background);
  height: 100%;
  padding: 0 1em;
  border-radius: $rounded-rectangle-radius;
  font-size: 1rem;

  @include createColoredCount($critical-selector, $gms-strong-warning);
  @include createColoredCount($warning-selector, $gms-warning);
  @include createColoredCount($info-selector, $gms-good-count);

  &__entry {
    @include centerChildren('vertical');
    cursor: pointer;
    & + & {
      margin-left: 1.5em;
    }
    &--disabled .system-message-summary__count {
      color: var(--gms-soft, $gms-soft);
    }
  }

  &__count {
    font-family: $gms-mono;
    font-size: 1.5rem;
    font-weight: bold;
    margin-left: 0.5ch;
    &--prefix {
      opacity: 0.33;
    }
    &--main {
      opacity: 1;
    }
  }
}

/// AG Grid Overrides
.system-message-table {
  .ag-body-viewport,
  .ag-root-wrapper {
    scroll-behavior: smooth;
  }
  .ag-row .ag-cell {
    background-color: var(
      system-message-prominent-background,
      var(--gms-prominent-background, $gms-prominent-background)
    );
    border-left: 1px solid var(--gms-prominent-background, $gms-prominent-background);
    border-right: 1px solid var(--gms-prominent-background, $gms-prominent-background);
  }
  .ag-header-cell {
    padding: 0 10px;
    background-color: var(--gms-popover-background, $gms-popover-background);
    .ag-header-cell-text {
      color: var(--gms-main, $gms-main);
    }
  }
  .ag-header-cell-label,
  .ag-header-cell-label {
    padding-left: 7px;
    font-size: 1rem;
    font-weight: 500;
  }
}
