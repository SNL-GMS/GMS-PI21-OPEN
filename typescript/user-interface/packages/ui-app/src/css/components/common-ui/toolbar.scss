.toolbar {
  margin-bottom: 0;
}
.toolbar + * {
  margin-top: 0.5rem;
}

.toolbar + .display-title {
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 0;
}
