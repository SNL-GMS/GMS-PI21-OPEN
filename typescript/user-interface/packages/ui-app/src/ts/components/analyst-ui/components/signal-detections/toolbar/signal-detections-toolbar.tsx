import { CheckboxDropdownToolbarItem, Toolbar } from '@gms/ui-core-components';
import type { AppDispatch } from '@gms/ui-state';
import {
  DisplayedSignalDetectionConfigurationEnum,
  SignalDetectionColumn,
  signalDetectionsActions,
  useAppDispatch,
  useAppSelector
} from '@gms/ui-state';
import Immutable from 'immutable';
import React from 'react';

import type { SignalDetectionsToolbarProps } from '~analyst-ui/components/signal-detections/types';
import {
  signalDetectionColumnDisplayStrings,
  signalDetectionSyncDisplayStrings,
  signalDetectionSyncLabelStrings,
  signalDetectionSyncRenderDividers
} from '~analyst-ui/components/signal-detections/types';
import { useBaseDisplaySize } from '~common-ui/components/base-display/base-display-hooks';

const BASE_DISPLAY_PADDING = 40;

export const onChangeHandler = (reduxDispatch: AppDispatch) => (event: any): void => {
  const eventObjectString = JSON.stringify(event);
  reduxDispatch(
    signalDetectionsActions.updateDisplayedSignalDetectionConfiguration(
      JSON.parse(eventObjectString)
    )
  );
};

// eslint-disable-next-line react/function-component-definition
export const SignalDetectionsToolbar: React.FunctionComponent<SignalDetectionsToolbarProps> = (
  props: SignalDetectionsToolbarProps
) => {
  const [displayWidthPx] = useBaseDisplaySize();
  const dispatch = useAppDispatch();
  const displayedSignalDetectionConfigurationObject = useAppSelector(
    state => state.app.signalDetections.displayedSignalDetectionConfiguration
  );

  // Map of values to pass to the checkbox dropdown toolbar item
  const displayedSignalDetectionConfigurationMap: Immutable.Map<
    DisplayedSignalDetectionConfigurationEnum,
    boolean
  > = React.useMemo(() => Immutable.fromJS(displayedSignalDetectionConfigurationObject), [
    displayedSignalDetectionConfigurationObject
  ]);

  const { selectedSDColumnsToDisplay, setSelectedSDColumnsToDisplay } = props;

  const itemsLeft: JSX.Element[] = React.useMemo(
    () => [
      <CheckboxDropdownToolbarItem
        key="SDdetections"
        label="Show detections"
        menuLabel="Show detections"
        tooltip="Display signal detections within the queried workflow time range."
        cyData="sd-table-detections-toggle"
        values={displayedSignalDetectionConfigurationMap}
        enumOfKeys={DisplayedSignalDetectionConfigurationEnum}
        enumKeysToDisplayStrings={signalDetectionSyncDisplayStrings}
        enumKeysToRenderDividers={signalDetectionSyncRenderDividers}
        enumKeysToLabelStrings={signalDetectionSyncLabelStrings}
        onChange={onChangeHandler(dispatch)}
      />,
      <CheckboxDropdownToolbarItem
        key="SDcolumns"
        label="Show columns"
        menuLabel="Show columns"
        tooltip="Select columns to be shown in the signal detection table below"
        onChange={setSelectedSDColumnsToDisplay}
        cyData="sd-table-column-picker"
        values={selectedSDColumnsToDisplay}
        enumOfKeys={SignalDetectionColumn}
        enumKeysToDisplayStrings={signalDetectionColumnDisplayStrings}
        itemSide="LEFT"
      />
    ],
    [
      dispatch,
      displayedSignalDetectionConfigurationMap,
      selectedSDColumnsToDisplay,
      setSelectedSDColumnsToDisplay
    ]
  );

  return (
    <Toolbar
      toolbarWidthPx={displayWidthPx}
      parentContainerPaddingPx={BASE_DISPLAY_PADDING}
      itemsLeft={itemsLeft}
    />
  );
};
