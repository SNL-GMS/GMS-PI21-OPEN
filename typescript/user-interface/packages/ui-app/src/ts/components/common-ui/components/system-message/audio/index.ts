export * from './system-message-alert-player';
