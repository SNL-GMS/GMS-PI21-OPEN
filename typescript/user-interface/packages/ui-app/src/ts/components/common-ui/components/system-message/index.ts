export { ReduxReactSystemMessageContainer as SystemMessage } from './system-message-container';
