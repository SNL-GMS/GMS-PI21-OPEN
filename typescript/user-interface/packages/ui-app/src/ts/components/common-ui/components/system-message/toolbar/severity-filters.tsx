import { SystemMessageTypes } from '@gms/common-model';
import { useForceUpdate } from '@gms/ui-util';
import Immutable from 'immutable';
import * as React from 'react';

export const buildDefaultSeverityFilterMap = (): Immutable.Map<
  SystemMessageTypes.SystemMessageSeverity,
  boolean
> => {
  let newMap = Immutable.Map<SystemMessageTypes.SystemMessageSeverity, boolean>();
  // Initialize the severity filter map
  Object.keys(SystemMessageTypes.SystemMessageSeverity).forEach(severity => {
    newMap = newMap.set(SystemMessageTypes.SystemMessageSeverity[severity], true);
  });
  return newMap;
};

export const useSeverityFilters = (): [
  Immutable.Map<SystemMessageTypes.SystemMessageSeverity, boolean>,
  React.Dispatch<
    React.SetStateAction<Immutable.Map<SystemMessageTypes.SystemMessageSeverity, boolean>>
  >
] => {
  const [severityFilterMap, setSeverityFilterMap] = React.useState<
    Immutable.Map<SystemMessageTypes.SystemMessageSeverity, boolean>
  >(undefined);
  const forceUpdate = useForceUpdate();
  React.useEffect(() => {
    const timeoutHandle = setTimeout(forceUpdate);

    const newMap = buildDefaultSeverityFilterMap();
    setSeverityFilterMap(newMap);

    return () => clearTimeout(timeoutHandle);
    // !FIX ESLINT Validate and check REACT HOOK dependencies
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return [severityFilterMap, setSeverityFilterMap];
};
