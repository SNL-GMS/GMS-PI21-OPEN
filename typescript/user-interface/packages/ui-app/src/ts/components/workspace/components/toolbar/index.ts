export { DeprecatedToolbar } from './toolbar-component';
