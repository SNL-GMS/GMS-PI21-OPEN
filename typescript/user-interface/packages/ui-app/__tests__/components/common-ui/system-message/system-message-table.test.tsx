import { uuid } from '@gms/common-util';
import noop from 'lodash/noop';
import * as React from 'react';

import { SystemMessageTable } from '../../../../src/ts/components/common-ui/components/system-message/system-message-table';
import { buildDefaultSeverityFilterMap } from '../../../../src/ts/components/common-ui/components/system-message/toolbar/severity-filters';
import type { SystemMessageTableProps } from '../../../../src/ts/components/common-ui/components/system-message/types';
import { systemMessages } from './shared-data';

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

// eslint-disable-next-line import/no-deprecated
const lodash = jest.requireActual('lodash');
lodash.uniqueId = () => '1';

let idCount = 0;
// eslint-disable-next-line no-plusplus
uuid.asString = jest.fn().mockImplementation(() => ++idCount);

describe('System Message Table', () => {
  it('should be defined', () => {
    expect(SystemMessageTable).toBeDefined();
  });

  const severityFilterMap = buildDefaultSeverityFilterMap();

  const getTimeColumnApiFunction = () => ({
    getColumnState: jest.fn(() => [
      {
        colId: 'time',
        sort: 'asc'
      }
    ])
  });

  const props: SystemMessageTableProps = {
    addSystemMessages: jest.fn(),
    clearAllSystemMessages: jest.fn(),
    isAutoScrollingEnabled: true,
    setIsAutoScrollingEnabled: jest.fn(),
    systemMessages,
    severityFilterMap
  };

  // eslint-disable-next-line react/jsx-props-no-spreading
  const table = Enzyme.shallow(<SystemMessageTable {...props} />);
  it('should match a snapshot', () => {
    expect(table).toMatchSnapshot();
  });

  const instance = table.instance();

  instance.state = { hasUnseenMessages: true };

  it('should not sort if props change if auto scrolling is enabled and sorted by time', () => {
    instance.forceUpdate = jest.fn();
    instance.table = {
      getColumnApi: getTimeColumnApiFunction,
      getTableApi: () => ({
        ensureIndexVisible: jest.fn(() => true),
        refreshInfiniteCache: jest.fn()
      })
    };

    instance.sortByTime = jest.fn();
    instance.componentDidUpdate(
      {
        isAutoScrollingEnabled: false
      },
      {
        systemMessages
      }
    );
    clearTimeout(instance.autoScrollTimer);
    expect(instance.sortByTime).not.toHaveBeenCalled();
  });

  it('should sort if props change if auto scrolling is enabled', () => {
    instance.table = {
      getColumnApi: () => ({
        getColumnState: jest.fn(() => [
          {
            colId: 'message',
            sort: 'asc'
          }
        ])
      }),
      getTableApi: () => ({
        ensureIndexVisible: jest.fn(() => true),
        refreshInfiniteCache: jest.fn()
      })
    };
    instance.sortByTime = jest.fn();
    instance.componentDidUpdate(
      {
        isAutoScrollingEnabled: false
      },
      {
        systemMessages
      }
    );
    expect(instance.sortByTime).toHaveBeenCalled();
  });

  it('can tell if it has scrolled to the end', () => {
    instance.getScrollContainer = () => ({
      scrollTop: 100, // how far from the top we have scrolled
      scrollHeight: 1000, // simulated height of whole container
      clientHeight: 900 // simulated amount visible
    });
    expect(instance.isScrolledToBottom()).toBeFalsy();
  });

  it('can tell if it has NOT scrolled to the end', () => {
    instance.getScrollContainer = () => ({
      scrollTop: 0, // how far from the top we have scrolled
      scrollHeight: 1000, // simulated height of whole container
      clientHeight: 900 // simulated amount visible
    });
    expect(instance.isScrolledToBottom()).toBeFalsy();
  });

  it('gets the right subcategory values', () => {
    const params = {
      data: {
        id: systemMessages[0].id
      }
    };
    const result = instance.subcategoryValueGetter(params);
    expect(result).toEqual(systemMessages[0].subCategory);
  });

  it('gets the right severity values', () => {
    const params = {
      data: {
        id: systemMessages[0].id
      }
    };
    const result = instance.severityValueGetter(params);
    expect(result).toEqual(systemMessages[0].severity);
  });

  it('gets the right message values', () => {
    const params = {
      data: {
        id: systemMessages[0].id
      }
    };
    const result = instance.messageValueGetter(params);
    expect(result).toEqual(systemMessages[0].message);
  });

  it('can checked if scrolled to start', () => {
    const result = instance.isScrolledToTop();
    expect(result).toBeTruthy();
  });
  it('can handle sort changed', () => {
    instance.onSortChanged();
    expect(instance.props.setIsAutoScrollingEnabled).toHaveBeenCalledTimes(1);
  });

  it('can tell scroll direction down', () => {
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    instance.state.prevScrollPositions = [0, 10];
    expect(instance.getScrollDirection()).toEqual('down');
  });

  it('can tell scroll direction up', () => {
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    instance.state.prevScrollPositions = [Number(instance.SCROLL_UP_THRESHOLD_PX) + 1, 0];
    expect(instance.getScrollDirection()).toEqual('up');
  });

  it('can tell scroll direction unchanged', () => {
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    instance.state.prevScrollPositions = [10, 10];
    expect(instance.getScrollDirection()).toEqual('unchanged');
  });

  it('can use new messages indicator', () => {
    instance.table = {
      getColumnApi: getTimeColumnApiFunction,
      getTableApi: () => ({
        ensureIndexVisible: jest.fn(() => true)
      })
    };
    instance.onNewMessageIndicatorClick();
    expect(instance.props.setIsAutoScrollingEnabled).toHaveBeenCalledWith(true);
  });

  it('can use on body scroll', () => {
    jest.useFakeTimers();
    instance.props.setIsAutoScrollingEnabled(true);
    instance.getScrollDirection = jest.fn().mockImplementation(() => 'up');
    instance.onBodyScroll();
    jest.runAllTimers();
    expect(instance.getScrollDirection).toHaveBeenCalled();
    expect(instance.props.setIsAutoScrollingEnabled).toHaveBeenLastCalledWith(false);
  });

  it('can scroll to latest', () => {
    jest.useFakeTimers();
    jest.spyOn(global, 'setTimeout');
    instance.scrollToLatest();
    expect(setTimeout).toHaveBeenCalledTimes(1);
  });

  // TODO Unskip tests
  it.skip('can attempt to scroll if rows are temporarily unavailable', () => {
    jest.useFakeTimers();
    jest.spyOn(global, 'setTimeout');
    instance.timeout();
    expect(setTimeout).toHaveBeenCalledTimes(1);
  });

  // TODO Unskip tests
  it.skip('will not attempt to scroll if rows are temporarily unavailable if timeout retries is zero', () => {
    jest.useFakeTimers();
    jest.spyOn(global, 'setTimeout');
    instance.timeout();
    expect(setTimeout).toHaveBeenCalledTimes(1);
    instance.SET_TIMEOUT_RETRIES = 0;
    jest.runOnlyPendingTimers();
    expect(setTimeout).toHaveBeenCalledTimes(1);
  });

  it('will not scroll to latest if there are no system messages', () => {
    const propsNoSystemMessages: SystemMessageTableProps = {
      addSystemMessages: jest.fn(),
      clearAllSystemMessages: jest.fn(),
      isAutoScrollingEnabled: true,
      setIsAutoScrollingEnabled: jest.fn(),
      systemMessages: undefined,
      severityFilterMap
    };

    // eslint-disable-next-line react/jsx-props-no-spreading
    const tableNoSystemMessages = Enzyme.mount(<SystemMessageTable {...propsNoSystemMessages} />);
    const instanceNoSystemMessages = tableNoSystemMessages.instance();
    jest.clearAllMocks();
    jest.useFakeTimers();
    jest.spyOn(global, 'setTimeout');
    instanceNoSystemMessages.scrollToLatest();
    expect(setTimeout).not.toHaveBeenCalled();
  });

  it('will not attempt to scroll if auto scrolling is not enabled', () => {
    const propsAutoScrollingDisabled: SystemMessageTableProps = {
      addSystemMessages: jest.fn(),
      clearAllSystemMessages: jest.fn(),
      isAutoScrollingEnabled: false,
      setIsAutoScrollingEnabled: jest.fn(),
      systemMessages: undefined,
      severityFilterMap
    };

    const tableAutoScrollingDisabled = Enzyme.mount(
      // eslint-disable-next-line react/jsx-props-no-spreading
      <SystemMessageTable {...propsAutoScrollingDisabled} />
    );
    const instanceAutoScrollingDisabled = tableAutoScrollingDisabled.instance();
    jest.clearAllMocks();
    jest.useFakeTimers();
    jest.spyOn(global, 'setTimeout');
    instanceAutoScrollingDisabled.timeout();
    expect(setTimeout).not.toHaveBeenCalled();
  });

  it('can create an empty datasource', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const table1 = Enzyme.mount(<SystemMessageTable {...props} />);
    const instance1 = table1.instance();
    const emptyDataSource = instance1.table.getEmptyDataSource();
    expect(emptyDataSource).toBeDefined();
  });

  it('can be destroyed', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const table2 = Enzyme.mount(<SystemMessageTable {...props} />);
    const instance2 = table2.instance();
    instance2.table.coreTableRef.getTableApi = () => ({
      purgeInfiniteCache: jest.fn(),
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      setDatasource: jest.fn(dataSource => noop)
    });
    instance2.table.destroy();
    expect(instance2.table.coreTableRef).toBeUndefined();
  });
});
