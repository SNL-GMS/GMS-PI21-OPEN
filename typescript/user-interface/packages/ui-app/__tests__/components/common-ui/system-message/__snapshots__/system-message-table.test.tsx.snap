// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`System Message Table should match a snapshot 1`] = `
<div
  className="system-message-table"
  data-cy="system-message-table"
  id="system-message-table"
>
  <Component>
    <InfiniteTable
      alwaysShowVerticalScroll={true}
      columnDefs={
        Array [
          Object {
            "colId": "id",
            "field": "id",
            "headerName": "ID",
            "headerTooltip": "ID",
            "hide": true,
            "width": 10,
          },
          Object {
            "cellRendererFramework": [Function],
            "colId": "time",
            "headerName": "Timestamp",
            "headerTooltip": "Timestamp",
            "sort": "asc",
            "valueGetter": [Function],
            "width": 200,
          },
          Object {
            "cellRendererFramework": [Function],
            "colId": "category",
            "comparator": [Function],
            "headerName": "Category",
            "headerTooltip": "Category",
            "valueGetter": [Function],
            "width": 110,
          },
          Object {
            "cellRendererFramework": [Function],
            "colId": "subCategory",
            "comparator": [Function],
            "headerName": "Subcategory",
            "headerTooltip": "Subcategory",
            "valueGetter": [Function],
            "width": 150,
          },
          Object {
            "cellRendererFramework": [Function],
            "colId": "severity",
            "comparator": [Function],
            "headerName": "Severity",
            "headerTooltip": "Severity",
            "valueGetter": [Function],
            "width": 105,
          },
          Object {
            "cellRendererFramework": [Function],
            "colId": "message",
            "comparator": [Function],
            "flex": 1,
            "headerName": "Message",
            "headerTooltip": "Message",
            "minWidth": 200,
            "valueGetter": [Function],
          },
        ]
      }
      debug={false}
      defaultColDef={
        Object {
          "cellRendererFramework": [Function],
          "disableStaticMarkupForCellRendererFramework": true,
          "disableStaticMarkupForHeaderComponentFramework": true,
          "filter": false,
          "headerComponentParams": Object {
            "template": "
  <div class=\\"ag-cell-label-container\\" role=\\"presentation\\">
      <span ref=\\"eMenu\\" class=\\"ag-header-icon ag-header-cell-menu-button\\"></span>
      <div ref=\\"eLabel\\" class=\\"ag-header-cell-label\\" role=\\"presentation\\">
          <span ref=\\"eText\\" class=\\"ag-header-cell-text\\" role=\\"columnheader\\"></span>
          <span ref=\\"eFilter\\" class=\\"ag-header-icon ag-filter-icon\\"></span>
          <!-- <span ref=\\"eSortOrder\\" class=\\"ag-header-icon ag-sort-order\\"></span> -->
          <span ref=\\"eSortAsc\\" class=\\"ag-header-icon ag-sort-ascending-icon\\"></span>
          <span ref=\\"eSortDesc\\" class=\\"ag-header-icon ag-sort-descending-icon\\"></span>
          <span ref=\\"eSortNone\\" class=\\"ag-header-icon ag-sort-none-icon\\"></span>
      </div>
  </div>
",
          },
          "sortable": true,
        }
      }
      headerHeight={48}
      onBodyScroll={[Function]}
      onCellClicked={[Function]}
      onCellContextMenu={[Function]}
      onGridReady={[Function]}
      onSortChanged={[Function]}
      overlayNoRowsTemplate="No System Messages"
      pagination={false}
      rowData={
        Array [
          Object {
            "category": "SOH",
            "id": "mock1",
            "message": "mock message text",
            "severity": "CRITICAL",
            "subCategory": "STATION",
            "time": 1593030714074,
            "type": "STATION_NEEDS_ATTENTION",
          },
          Object {
            "category": "SOH",
            "id": "mock2",
            "message": "mock message text",
            "severity": "CRITICAL",
            "subCategory": "STATION",
            "time": 1593030714075,
            "type": "STATION_NEEDS_ATTENTION",
          },
        ]
      }
      rowHeight={40}
      suppressCellFocus={true}
      suppressScrollOnNewData={true}
    />
  </Component>
  <NewMessageIndicator
    handleNewMessageIndicatorClick={[Function]}
    isVisible={false}
  />
</div>
`;
